Ve.Messaging/Model/Message.cs

```diff
@@ -1,23 +1,37 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.IO;
 
 namespace Ve.Messaging.Model
 {
     public class Message
     {
+        private readonly Action _complete;
+
         public Message(Stream bodyStream,
                        string sessionId = "",
                        string label = "",
-                       IDictionary<string, object> properties = null)
+                       string id = "",
+                       IDictionary<string, object> properties = null,
+                       Action complete = null)
         {
             BodyStream = bodyStream;
+            Id = id;
             SessionId = sessionId;
             Label = label;
             Properties = properties;
-        }
+            _complete = complete;
+        }
+
+        public string Id { get; }
         public Stream BodyStream { get; }
         public string Label { get; }
         public string SessionId { get; }
-        public IDictionary<string, object> Properties { get; } 
+        public IDictionary<string, object> Properties { get; }
+
+        public Action Complete
```

Comment:
but using the action means that the .net runtime will need the library anyway, because the message itself will be referenced by the closure on the action